### DISCUSSION TITLE
How does `binarySource` work / can it be configured per-package-manager?

### DISCUSSION THREAD

**Discussion by stdedos** [renovatebot/renovate#27867] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, latest

### Please tell us more about your question or problem

I need to use a "custom-cooked" script for `bundler`. However, if I'd like to be able support other environments (e.g. npm, pip, etc) by Renovate fetching its tools "as per defaults".

The only thing close to what I want looks like https://docs.renovatebot.com/self-hosted-configuration/#binarysource.

However, I am confused on how would that work.
1. `binarySource=global` means that ALL of the tools need to be available on PATH. They are not
2. `binarySource=docker` means that I need to build a giant Docker image including all the tools. Not an option
3. We don't use Hermit 😕 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
Use default install mode and set bundler to ignore[^1]

You can use `env`[^2] option (needs setting `allowedEnv`[^3] too)
```json
{
  "env": {
    "IGNORED_TOOLS": "bundler"
  }
}
```

[^1]: https://github.com/containerbase/base?tab=readme-ov-file#temporary-disable-tool-installer
[^2]: https://docs.renovatebot.com/configuration-options/#env
[^3]: https://docs.renovatebot.com/self-hosted-configuration/#allowedEnv

**Reply by stdedos** [reply]
I don't have a `config.js` file - nor know how to get one.
If the option is to commit it to the repository-in-check, I'd rather not
If it's "our renovate repository", it cannot be - since we are mirroring https://gitlab.com/renovate-bot/renovate-runner

How would `RENOVATE_ALLOWED_ENV` be configured "as a `array[str]`"? 😕

**Reply by stdedos** [reply]
Also - how is `RENOVATE_ALLOWED_ENV` expected to work? 😕 

I have added:

```yaml
renovate:
  variables:
    RENOVATE_ALLOWED_ENV: "IGNORED_TOOLS"
```

But the `renovate-config-validator` actively complains.

Also, on a learning tangent: Why `renovate-config-validator` and `renovate` do "different things"?
I hear the benefit of "running no matter what", but I prefer things to be as centralized as possible.

**Reply by viceice** [reply]
why are you mirror our runner repo? that's wrong. you should configure a new repo and reference our templates.

**Reply by stdedos** [reply]
I reference the templates of the mirrored repo (https://gitlab.com/%mirror%/renovate-runner)

ik it's not the same, but the end-result (Merged YAML) is the same

**Reply by viceice** [reply]
simply add a config.js to your runner repo root and configure needed options there.

I'm not sure but array options must be valid JSON arrays.

please provide new debug logs for the bundler call. you probably need to check the force rebase checkbox to get a new commit and bundler run.
